src/audio/module_adapter/module_adapter.c

```diff
@@ -175,37 +177,33 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 
 static int module_adapter_sink_src_prepare(struct comp_dev *dev)
 {
-	struct sof_sink *audio_sink[PLATFORM_MAX_STREAMS];
-	struct sof_source *audio_src[PLATFORM_MAX_STREAMS];
 	struct processing_module *mod = comp_get_drvdata(dev);
 	struct list_item *blist;
-	uint32_t num_of_sources = 0;
-	uint32_t num_of_sinks = 0;
 	int ret;
+	int i;
 
 	/* acquire all sink and source buffers, get handlers to sink/source API */
+	i = 0;
 	list_for_item(blist, &dev->bsink_list) {
-		struct comp_buffer *sink_buffer;
-
-		sink_buffer = container_of(blist, struct comp_buffer, source_list);
-		audio_sink[num_of_sinks] =
-				audio_stream_get_sink(&sink_buffer->stream);
-		sink_reset_num_of_processed_bytes(audio_sink[num_of_sinks]);
-		num_of_sinks++;
+		struct comp_buffer *sink_buffer =
+				container_of(blist, struct comp_buffer, source_list);
+		mod->sinks[i] = audio_stream_get_sink(&sink_buffer->stream);
+		i++;
```

Comment:
It won't always point to buffers - it may be to dpQueue, LL shadred queue, DAI (dai does not need to use any buffer - it may just expose source API), etc. etc.

Reply:
another reason - performance
call to the module looks like: 

```
	int (*process)(struct processing_module *mod,
		       struct sof_source **sources, int num_of_sources,
		       struct sof_sink **sinks, int num_of_sinks);
```

just a simple pass of pointers

Reply:
arrays are certainly faster when accessing random elements - it's faster to find the 17th element in an array than to iterate 17 list entries. But (1) we hardly ever have more than 2 buffers in those arrays / lists, and (2) we usually scan them sequentially. In that case I'm not sure arrays are (much) faster. And maintaining multiple access ways to the same data is often a significant burden.